src/equations.jl

```diff
@@ -48,6 +48,13 @@ Base.:~(lhs::Number    , rhs::Num) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Symbolic, rhs::Symbolic) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Symbolic, rhs::Any    ) = Equation(value(lhs), value(rhs))
 Base.:~(lhs::Any, rhs::Symbolic    ) = Equation(value(lhs), value(rhs))
+for T in [:Num, :Complex, :Number], S in [:Num, :Complex, :Number]
```

Comment:
Do we want to allow Number ~ Number?

Reply:
The `(T != :Complex && S != :Complex) && continue` takes care of allowing only Complex with something else.